src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,571 @@
+
+        expect(player.setMaxVideoBitrate(500)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(500);
+
+        expect(player.setMaxVideoBitrate(3)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(3);
+
+        expect(player.setMaxVideoBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxVideoBitrate(oldMax)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(oldMax);
+      });
+    });
+
+    describe("setMaxAudioBitrate/getMaxAudioBitrate", () => {
+      /* tslint:disable:max-line-length */
+      it("should update the maximum audio bitrate when calling setMaxAudioBitrate by default", () => {
+      /* tslint:enable:max-line-length */
+        const oldMax = player.getManualAudioBitrate();
+
+        expect(player.setMaxAudioBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxAudioBitrate(500)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(500);
+
+        expect(player.setMaxAudioBitrate(3)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(3);
+
+        expect(player.setMaxAudioBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxAudioBitrate(oldMax)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(oldMax);
+      });
+    });
+
+    describe("getMaxBufferBehind/setMaxBufferBehind", () => {
+      /* tslint:disable:max-line-length */
+      it("should update the max buffer behind through setMaxBufferBehind by default", () => {
+      /* tslint:enable:max-line-length */
+        expect(player.setMaxBufferBehind(50)).to.equal(undefined);
+        expect(player.getMaxBufferBehind()).to.equal(50);
+
+        expect(player.setMaxBufferBehind(Infinity)).to.equal(undefined);
+        expect(player.getMaxBufferBehind()).to.equal(Infinity);
+      });
+    });
+
+    describe("getMaxBufferAhead/setMaxBufferAhead", () => {
+      /* tslint:disable:max-line-length */
+      it("should update the max buffer behind through setMaxBufferAhead by default", () => {
+      /* tslint:enable:max-line-length */
+        expect(player.setMaxBufferAhead(50)).to.equal(undefined);
+        expect(player.getMaxBufferAhead()).to.equal(50);
+
+        expect(player.setMaxBufferAhead(Infinity)).to.equal(undefined);
+        expect(player.getMaxBufferAhead()).to.equal(Infinity);
+      });
+    });
+
+    describe("getWantedBufferAhead/setWantedBufferAhead", () => {
+      it("should update the buffer goal through setWantedBufferAhead by default", () => {
+        expect(player.setWantedBufferAhead(50)).to.equal(undefined);
+        expect(player.getWantedBufferAhead()).to.equal(50);
+
+        expect(player.setWantedBufferAhead(Infinity)).to.equal(undefined);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this use of the output from "setWantedBufferAhead"; "setWantedBufferAhead" doesn't return anything. ([typescript:S3699](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS3699))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=414&issues=AWgOMLyH_MPaBDn0sxTV&open=AWgOMLyH_MPaBDn0sxTV)

[ISSUE_KEY]: AWgOMLyH_MPaBDn0sxTV